src/Centreon/Application/Controller/MonitoringResourceController.php

```diff
@@ -140,8 +140,17 @@ public function list(
 
         $context->addExclusionStrategy(new ResourceExclusionStrategy());
 
-        $resources = $this->resource->filterByContact($this->getUser())
-            ->findResources($filter);
+        // try to avoid exception from the regexp bad syntax in search criteria
+        try {
+            $resources = $this->resource->filterByContact($this->getUser())
+                ->findResources($filter);
+        } catch (\PDOException $e) {
+            if ($e->getCode() !== "42000") {
+                throw $e;
```

Comment:
The controller must not be aware of structure exceptions.
PDO is a class to read data so it's not a business class.

Reply:
Your repository can raised a specifi exception that will can be catch by the service

Reply:
could you give an example from the existing code base, because the idea is to don't change the behavior that we have already just to avoid the particular problem described in the PR

Reply:
ok